### ISSUE TITLE
Messaging/ Alert system 

### ISSUE THREAD

**Issue by aleincsharp** [input-output-hk/daedalus#2488]
Hello everybody,

as a small pool operator, I currently stumble across overcrowded pools every day and what is even worse:
Retired pools currently have a total of over 17 million ADAs delegated!

My idea was to implement a kind of messaging / alert system in Daedalus, where the delegator in the system receives a message,
that he is currently staked in a pool that is either overcrowded or that the pool is retired.

Would this idea be a possible approach to draw attention to it?

P.S .: I have already introduced myself to @nikolaglumac as a developer and I hope to be able to contribute soon! :)

Arthur (WISE-Pool)

**Comment by nikolaglumac** [comment]
Hi Arthur :-)

Can you expand on the idea? 
What messaging/alert type do you have in mind?
Also, we are limited to showing the alerts only once Daedalus is running.
E.g. if you delegate and then don't run Daedalus for a year (during which time the SP you delegate to retires), we have no way to notify the user.

More info is greately appreciated! 

Thanks!

**Comment by timack** [comment]
Thats an interesting idea ... but I wonder if this should be a 3rd party service maybe optionally initiated from within the wallet - ie. you could (optionally) provide your email and and pool id to a monitoring service who would email you if the pool over-saturates, goes down or gets retired. Then you you're not reliant on having to check the wallet to see if the pool is still active?

If I delegated and came back after a year to find out my pool had been retired for most of the previous year I'd be pretty mad.

But maybe such functionality is outside of the scope of the wallet's responsibility and the onus should be on the delegator to monitor their pool appropriately?

**Comment by aleincsharp** [comment]
> Hi Arthur :-)
> 
> Can you expand on the idea?
> What messaging/alert type do you have in mind?
> Also, we are limited to showing the alerts only once Daedalus is running.
> E.g. if you delegate and then don't run Daedalus for a year (during which time the SP you delegate to retires), we have no way to notify the user.
> 
> More info is greately appreciated!
> 
> Thanks!

Of course, you can only send a notification when the application is running.
First, a "warning message" could be displayed when someone wants to delegate to an overcrowded pool.
It would also be important that the delegator receives a notification as soon as a pool is classified as retired.

At the first step a in-app alert system would be good and maybe a desktop push notification.
Maybe a optional subscription from daedalus wallet with a email connection could be a possible second solution to reach delegators outside the application.

**Comment by aleincsharp** [comment]
> Thats an interesting idea ... but I wonder if this should be a 3rd party service maybe optionally initiated from within the wallet - ie. you could (optionally) provide your email and and pool id to a monitoring service who would email you if the pool over-saturates, goes down or gets retired. Then you you're not reliant on having to check the wallet to see if the pool is still active?
> 
> If I delegated and came back after a year to find out my pool had been retired for most of the previous year I'd be pretty mad.
> 
> But maybe such functionality is outside of the scope of the wallet's responsibility and the onus should be on the delegator to monitor their pool appropriately?

Yes, this was also in my mind to create a API and link the wallet to a email or similar.

**Comment by nikolaglumac** [comment]
All of this would require Daedalus to be constantly running and that is not realistic.
It is obviously a 3rd party service requirement that needs to be integrated with Daedalus.

As far as Daedalus goes the best we can do is to run a check to see if any of the wallets you have are delegated to a retired stake pool and raise an alert in case they are...

**Comment by aleincsharp** [comment]
> As far as Daedalus goes the best we can do is to run a check to see if any of the wallets you have are delegated to a retired stake pool and raise an alert in case they are...

This would be a great step!

**Comment by aleincsharp** [comment]
> All of this would require Daedalus to be constantly running and that is not realistic.
> It is obviously a 3rd party service requirement that needs to be integrated with Daedalus.

Do you have an idea how best to integrate with a 3rd party app to daedalus? A plugin, a api?

**Comment by nikolaglumac** [comment]
An API would do.

**Comment by Asafkbalink** [comment]
@aleincsharp [PoolTool](https://play.google.com/store/apps/details?id=com.pegasus.tool&hl=en&gl=US) on android can provide you updates if the pool changed fees, became over-saturated or produced a new block. 
Regardless, this feature is important to have in Daedalus as well. Daedalus have wider adoption

**Comment by aleincsharp** [comment]
> @aleincsharp [PoolTool](https://play.google.com/store/apps/details?id=com.pegasus.tool&hl=en&gl=US) on android can provide you updates if the pool changed fees, became over-saturated or produced a new block.
> Regardless, this feature is important to have in Daedalus as well. Daedalus have wider adoption

Yes, the pooltool and adapools apps can do this. 
I am working currently on the API! Because this API i a part of my bigger platform/solution it takes more time for now. But i hope to get the first version online soon!

**Comment by timack** [comment]
This incident might have been the sort of thing a delegator would have wanted to know about - Pool operator increased margin to 99% just before the end of Epoch: 
https://twitter.com/SproutPool/status/1390036177407725568

**Comment by aleincsharp** [comment]
> This incident might have been the sort of thing a delegator would have wanted to know about - Pool operator increased margin to 99% just before the end of Epoch:
> https://twitter.com/SproutPool/status/1390036177407725568

Yes, i am working on multiple notifications through telegram, email, twitter, etc.
Because this project/API is a part of my new plattform (https://ada-data.eu/), it takes a little bit longer to implement.

**Comment by nikolaglumac** [comment]
@aleincsharp is that project of yours publicly available? AKA. open source?

**Comment by aleincsharp** [comment]
> @aleincsharp is that project of yours publicly available? AKA. open source?

At the moment no, but i plan to do it, yes!

There is a lot to do, but unfortunately I don't have that much spare time with a normal job. I do it like many others in my spare time. I'll do my best and hurry up! :)

**Comment by nikolaglumac** [comment]
Thanks for all the efforts @aleincsharp!
We need to open more direct communication in order to integrate your service within Daedalus.
Perhaps you could prepare an architectural overview or a plan and present it to us in a call?
We would be glad to push this forward in the 2nd part of May.
cc @darko-mijic

**Comment by nikolaglumac** [comment]
I meant like Google Meet session.

**Comment by aleincsharp** [comment]
> I meant like Google Meet session.

We can schedule the meeting on any platform. As it suits you best.

**Comment by aleincsharp** [comment]
> Thanks for all the efforts @aleincsharp!
> We need to open more direct communication in order to integrate your service within Daedalus.
> Perhaps you could prepare an architectural overview or a plan and present it to us in a call?
> We would be glad to push this forward in the 2nd part of May.
> cc @darko-mijic

Hello @nikolaglumac ,

i am ready to show some pics about the architecture and share my ideas with you;). When you can schedule a meeting?

**Comment by nikolaglumac** [comment]
Hi @aleincsharp! That's great to hear! 
I'll reach out to you near the end of next week and we'll find a slot which works for all of us :-) 
Looking forward to it!

**Comment by nikolaglumac** [comment]
Hey @aleincsharp! 👋 
I haven't forgotten about you - we've just been very busy and Darko is on vacation this week.
This is why I would like to ask you to move our call to next week (Wednesday or Thursday).
Let's continue via email - please reach on nikola.glumac@iohk.io
Thanks!

**Comment by hodlonaut** [comment]
Has anything happened on this front since 20th of May? Just wondering...

**Comment by aleincsharp** [comment]
> Has anything happened on this front since 20th of May? Just wondering...

Yes, working again on it after a forced break...will try to release first steps soon.

**Comment by disassembler** [comment]
Closing as part of issue backlog cleanup — issues older than 90 days are being retired to give the new team a clean slate. Please open a new issue if this is still relevant.
